presto-main/src/test/java/com/facebook/presto/operator/window/TestLagFunction.java

```diff
@@ -190,5 +191,33 @@ public void testLagFunction()
                         .row(32, "O", "1995-07-16")
                         .row(34, "O", "1998-07-21")
                         .build());
+
+        assertWindowQueryWithNulls("lag(orderkey, 1, -1) IGNORE NULLS OVER (PARTITION BY orderstatus ORDER BY orderkey)",
+                resultBuilder(TEST_SESSION, BIGINT, VARCHAR, BIGINT)
+                        .row(3L, "F", -1L)
+                        .row(5L, "F", 3L)
+                        .row(6L, "F", 5L)
+                        .row(null, "F", 6L)
+                        .row(34L, "O", -1L)
+                        .row(null, "O", 34L)
+                        .row(1L, null, -1L)
+                        .row(7L, null, 1L)
+                        .row(null, null, 7L)
+                        .row(null, null, 7L)
+                        .build());
+
+        assertWindowQueryWithNulls("lag(orderstatus, 1, null) IGNORE NULLS OVER (ORDER BY orderkey, orderstatus)",
+                resultBuilder(TEST_SESSION, BIGINT, VARCHAR, VARCHAR)
+                        .row(1L, null, null)
+                        .row(3L, "F", null)
+                        .row(5L, "F", "F")
+                        .row(6L, "F", "F")
+                        .row(7L, null, "F")
+                        .row(34L, "O", "F")
+                        .row(null, "F", "O")
+                        .row(null, "O", "F")
+                        .row(null, null, "O")
+                        .row(null, null, "O")
+                        .build());
     }
```

Comment:
Please add tests with `RESPECT NULLS` explicitly set, in addition to `IGNORE NULLS` tests.